### PR TITLE
feat: external profile

### PR DESCRIPTION
**Pull request by jonowu**
## Purpose

We should be able to view a person's profile if we tap on their name/picture in the leaderboard or social feed.

## Changes

- Moved the visualisation into its own screen (Profile Stats). Reason I did this was the keep the profile screen clean, and allow us to reuse the profile screen for viewing other people's profiles.
- In the profile screen, check if a externalUserId prop is received, if so, use that to query the profile bio, otherwise, use the logged in user's id
- only show the buttons (logout etc.) if not an external user

### Screenshots
96121238-8263d580-0f3b-11eb-96a6-066d32fb533d

## Checklist

- [ ] Tested on Android
- [ ] Tested on iOS

### CODE REVIEW

File: screens/profile/index.js [hunk 1]
```diff
@@ -217,23 +184,25 @@ function ProfileScreen({ navigation }) {
...
-                      }
-                    >
-                      <MaterialCommunityIcons name={item.iconName} size={35} color="black" />
-                      <Body style={{ fontSize: 17, fontWeight: 'bold', left: 20 }}>{item.title}</Body>
-                    </ProfileButton>
-                  ))}
-                </ProfileButtonsContainer>
-              )}
-            </AuthenticatedContext.Consumer>
+            {!externalUserId && (
```
Comment (BenjaminYong) [inline screens/profile/index.js:187]: Genius - what a smart way to reuse the profile 💯

File: screens/profile-stats/index.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import { ButtonGroup } from 'react-native-elements';
+import { gql, useQuery } from '@apollo/client';
+import { useIsFocused } from '@react-navigation/native';
+
+import { AuthenticatedContext } from '../../context/authenticated-context';
+import { sdgs } from '../../data/sdgs';
+import Screen from '../../components/screen';
+import Visualisation from '../../components/visualisation';
+
+function ProfileStatsScreen({ navigation }) {
```
Comment (BenjaminYong) [inline screens/profile-stats/index.js:13]: I agree, I think its a good idea to move the visualisation into its own stats screen. Adds space for future potential detailed stats, badges, and other details 👍🏼

File: screens/profile-stats/index.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    const visualisationData = {
+      totalSdgCount: totalSdgCount,
+      userSdgCount: userSdgCount,
+      totalColors: totalColors,
+      userColors: userColors,
+    };
+
+    return (
+      <Screen centeredHorizontally>
+        <ButtonGroup
```
Comment (BenjaminYong) [inline screens/profile-stats/index.js:102]: I like the consistent styling of button group in stats vs feed. Should we sync up the colours to a Swinburne branded colour? I reckon the yellow or red will look good

File: screens/leaderboard/index.js [hunk 4]
```diff
@@ -141,31 +128,38 @@ function PodiumItem({ data, index }) {
...
+        size="medium"
+        onPress={() => navigation.navigate({ name: 'User Profile', params: { externalUserId: data.id } })}
+      />
       <ItemTextContainer>
-        <ItemName>
-          <Body variant={2} adjustsFontSizeToFit numberOfLines={1} style={{ fontWeight: 'bold' }}>
+        <TouchableOpacity
+          onPress={() => navigation.navigate({ name: 'User Profile', params: { externalUserId: data.id } })}
+        >
+          <Body
```
Comment (BenjaminYong) [inline screens/leaderboard/index.js:151]: Thanks for the whole refactor on the visualisation components using typography ✌🏼

**Comment by BenjaminYong** [comment]
BLT mate! Well done 😄 🥪 
Just the comment on the colours for the button group, but everything else works great.
Smart move on reusing the profile component to show these bios, low effort big results 💣

File: screens/profile-stats/index.js (hunk 3 above)
Comment (jonowu) [reply screens/profile-stats/index.js:102]: Good idea, pushed up a change for it:
Screen Shot 2020-10-16 at 7 20 50 pm
Screen Shot 2020-10-16 at 7 20 57 pm

**Review by BenjaminYong — APPROVED** [review]
Great stuff man, loving the Swin yellow buttons
👍🏼

**Pull request merged**